package.json

```diff
@@ -63,7 +64,12 @@
   },
   "lint-staged": {
```

Comment:
General question: is this absolutely required? I know there are lots of repos that do this, but (for me personally) it wastes a lot of time because I have to wait for lint to run until I can commit + push to remotes and it makes me not want to commit because there's too much friction

Reply:
I agree. I would prefer a pre-push hook rather than pre-commit

Reply:
With CI, failing lint prevents merge

Reply:
But CI is after pushing changes, right? And CI evaluates the last commit.

Reply:
Something like this: https://github.com/theenadayalank/lint-prepush?

Reply:
Why can't we just do this:
```
"pre-push": "npm run lint:fix"
```

Reply:
If the task returns without any error (even if it linted some files, unstaged so), it will continue to push without creating a fix commit.

Reply:
> The `lint-staged` task runs really fast on my computer, is it slow for you?

Yep, adds 3-5 seconds, even though I have an ssd 😭

Reply:
I usually see projects checking for linting errors only on CI. If you decide on going in this route, I think it's preferable to have linting done automatically on push instead of on every commit, since the latter makes the entire iteration process slower and more annoying.
Regarding creating a `lint-fix` commit automatically on push / commit, that doesn't seem like a good idea to me, since many times I've had auto-fixable linting errors which I needed to fix manually as that was more appropriate.

These are just my opinions and observations from working on other projects, though.